src/jdk.jlink/share/classes/jdk/tools/jlink/internal/Platform.java

```diff
@@ -24,12 +24,31 @@
  */
 package jdk.tools.jlink.internal;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteOrder;
+import java.util.HashMap;
 import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
 
 /**
  * Supported platforms
  */
-public record Platform(OperatingSystem os, Architecture arch) {
+public record Platform(OperatingSystem os, Architecture arch, ByteOrder endianness) {
+    private static final Properties KNOWN_ENDIANNESS;
```

Comment:
I assume this should be renamed to something like PLATFORM_PROPERTIES as the properties are not specific to endianness now.

Reply:
Hello Alan, I've updated the PR with this and the other test related suggestions.